Makefile

```
@@ -167,10 +209,27 @@ update-ui:
 	@printf $(COLOR) "Install/update temporal ui-server..."
 	@go install github.com/temporalio/ui-server/cmd/server@latest
 
-update-tools: update-goimports update-linters update-mockgen update-proto-plugins update-proto-linters update-gotestsum
+$(TEMPDIR) := $(shell mktemp -d)
```

these are never going to get cleaned up, are they? also I think you mean
```suggestion
TEMPDIR := $(shell mktemp -d)
```

but anyway, can we make a temp dir only when needed and clean it up afterwards?

---

Done. We now create and remove temporary directories only when installing new tools